diff/parse.go

```diff
@@ -59,7 +59,7 @@ func (r *MultiFileDiffReader) ReadFile() (*FileDiff, error) {
 		switch e := err.(type) {
 		case *ParseError:
 			if e.Err == ErrNoFileHeader || e.Err == ErrExtendedHeadersEOF {
-				return nil, io.EOF
+				return fd, io.EOF
```

Comment:
I know it's a bit unidiomatic to return something meaningful alongside an error, but on the other hand the functional change here is very small, and doesn't break existing clients in any way. The new behaviour is entirely optional.